### ISSUE TITLE
Rule Change: padded-blocks

### ISSUE THREAD

**Issue by josephzidell** [eslint/eslint#15130]
### What rule do you want to change?

padded-blocks

### What change to do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
declare namespace Types {

    type Foo = 'string';

}
```

### What does the rule currently do for this code?

Currently this rule checks blocks, classes, and switch statements. While technically this _is a block_, it is a different ASTNode.

### What will the rule do after it's changed?

Add an option "declares" which would require or disallow padding within declare statements

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

A `declare` statement is a kind of code block, yet since it is assigned a different AST Node, the `padded-blocks` rules does not pick this up as a violation.

This proposal would add an option `"declares"` and it would be used, have the same options, and act the same way as the other, existing options.

**Comment by mdjermanovic** [comment]
Hi @josephzidell, thanks for the issue! 

> This proposal would add an option `"declares"` and it would be used, have the same options, and act the same way as the other, existing options.

Core rules don't have typescript-specific options. You can ask for an [extension rule](https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin#extension-rules) in [typescript-eslint/typescript-eslint](https://github.com/typescript-eslint/typescript-eslint) repository.
